crates/ruff_workspace/src/resolver.rs

```diff
@@ -140,19 +144,31 @@ impl<'a> Resolver<'a> {
     }
 
     /// Add a resolved [`Settings`] under a given [`PathBuf`] scope.
-    fn add(&mut self, path: PathBuf, settings: Settings) {
-        self.settings.insert(path, settings);
+    fn add(&mut self, path: &Path, settings: Settings) {
+        self.settings.push(settings);
+
+        // normalize the path to use `/` separators and escape the '{' and '}' characters,
+        // which matchit uses for routing parameters
+        let path = path.to_slash_lossy().replace('{', "{{").replace('}', "}}");
+
+        match self
+            .router
+            .insert(format!("{}/{{*filepath}}", path), self.settings.len() - 1)
+        {
+            Ok(_) => {}
+            Err(InsertError::Conflict { .. }) => {}
+            Err(_) => unreachable!("file paths are escaped before being inserted in the router"),
```

Comment:
I was curious about this, and I think I convinced myself that this is correct because all other possible errors (aside from `Conflict`) can only occur as a result of characters that the router treats as special. And the escaping done above means that there should be precisely zero special characters in the path.

(It might make sense to add an `escape` routine to `matchit`. Similar to `regex::escape`. That way, you can provide more of a stronger guarantee here.)

Reply:
Yeah after escaping we guarantee that we're inserting a static route, so none of the other errors are possible. An `escape` function in `matchit` is a good idea. I was also considering optimizing the double `String::replace` but decided it wasn't worth it for us.